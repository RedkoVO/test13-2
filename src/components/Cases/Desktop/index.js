import React from 'react'
import PropTypes from 'prop-types'
import withStyles from '@material-ui/core/styles/withStyles'

import MenuNavigation from '../../MenuNavigation/Desktop/AsyncMenuNavigation'
import Header from '../../../containers/Header/AsyncHeader'
import Footer from '../../Footer/Desktop/'
import CasesFilter from './components/CasesFilter'
import CasesList from './components/CasesList'

import styles from './styles'

const Cases = ({ classes, handleLogout, cases }) => (
  <div className="mainWrapperTrade">
    <div className={classes.containerTradeWrap}>
      <div className={classes.containerTrade}>
        <header className={classes.headerSkins}>
          <MenuNavigation type="main" page="dark" handleLogout={handleLogout} />
          <Header type="catalog" />
        </header>
        <main className={classes.cases}>
          <CasesFilter />
          <CasesList cases={cases} />
        </main>
      </div>
    </div>
    <Footer />
  </div>
)

Cases.propTypes = {
  classes: PropTypes.object,
  cases: PropTypes.array
}

export default withStyles(styles)(Cases)
