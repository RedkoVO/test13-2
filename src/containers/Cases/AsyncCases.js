import Async from '../../components/App/components/Async'

export const AsyncCases = Async({
  loader: () => import(/* webpackChunkName: "cases_page" */ './index')
})

export default AsyncCases
