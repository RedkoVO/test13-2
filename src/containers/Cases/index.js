import compose from 'recompose/compose'
import { withProps, pure } from 'recompose'

import AsyncCasesDesktop from '../../components/Cases/Desktop/AsyncCasesDesktop'

import CaseXIcon from '../../assets/images/caseX.png'

const tmpCases = [
  {
    id: 1,
    img: CaseXIcon,
    title: 'Hello, buddy!',
    cost: '$34.20',
    game: 'CS:GO'
  },
  {
    id: 2,
    img: CaseXIcon,
    title: 'Hello, buddy!',
    cost: '$34.20',
    game: 'CS:GO'
  },
  {
    id: 3,
    img: CaseXIcon,
    title: 'Hello, buddy!',
    cost: '$34.20',
    game: 'CS:GO'
  },
  {
    id: 4,
    img: CaseXIcon,
    title: 'Hello, buddy!',
    cost: '$34.20',
    game: 'CS:GO'
  },
  {
    id: 5,
    img: CaseXIcon,
    title: 'Hello, buddy!',
    cost: '$34.20',
    game: 'CS:GO'
  },
  {
    id: 6,
    img: CaseXIcon,
    title: 'Hello, buddy!',
    cost: '$34.20',
    game: 'CS:GO'
  },
  {
    id: 7,
    img: CaseXIcon,
    title: 'Hello, buddy!',
    cost: '$34.20',
    game: 'CS:GO'
  },
  {
    id: 8,
    img: CaseXIcon,
    title: 'Hello, buddy!',
    cost: '$34.20',
    game: 'CS:GO'
  },
  {
    id: 9,
    img: CaseXIcon,
    title: 'Hello, buddy!',
    cost: '$34.20',
    game: 'CS:GO'
  },
  {
    id: 10,
    img: CaseXIcon,
    title: 'Hello, buddy!',
    cost: '$34.20',
    game: 'CS:GO'
  },
  {
    id: 11,
    img: CaseXIcon,
    title: 'Hello, buddy!',
    cost: '$34.20',
    game: 'CS:GO'
  },
  {
    id: 12,
    img: CaseXIcon,
    title: 'Hello, buddy!',
    cost: '$34.20',
    game: 'CS:GO'
  }
]

export default compose(
  withProps(() => ({ cases: tmpCases })),
  pure
)(AsyncCasesDesktop)
