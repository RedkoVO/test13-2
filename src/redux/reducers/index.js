import { combineReducers } from 'redux'
import { reducer as form } from 'redux-form'
import { APP_RESET } from '../actions/types'
import auth from './auth'
import balance from './balance'
import config from './config'
import history from './history'

const appReducer = combineReducers({
  form,
  auth,
  balance,
  config,
  history
})

export default (state, action) => {
  if (action.type === APP_RESET) {
    state = undefined
  }

  return appReducer(state, action)
}
