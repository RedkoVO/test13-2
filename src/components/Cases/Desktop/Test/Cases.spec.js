import * as React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import { createStore } from 'redux'

import Cases from '../'

import CaseXIcon from '../../../../assets/images/caseX.png'

const tmpCases = [
  {
    id: 1,
    img: CaseXIcon,
    title: 'Hello, buddy!',
    cost: '$34.20',
    game: 'CS:GO'
  },
  {
    id: 2,
    img: CaseXIcon,
    title: 'Hello, buddy!',
    cost: '$34.20',
    game: 'CS:GO'
  },
  {
    id: 3,
    img: CaseXIcon,
    title: 'Hello, buddy!',
    cost: '$34.20',
    game: 'CS:GO'
  },
  {
    id: 4,
    img: CaseXIcon,
    title: 'Hello, buddy!',
    cost: '$34.20',
    game: 'CS:GO'
  }
]
const store = createStore(a => a, {})

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter keyLength={0}>
          <Cases cases={tmpCases} />
        </MemoryRouter>
      </Provider>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
