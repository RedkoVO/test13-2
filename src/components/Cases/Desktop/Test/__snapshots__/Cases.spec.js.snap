// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="mainWrapperTrade"
>
  <div
    className="Cases-containerTradeWrap-1"
  >
    <div
      className="Cases-containerTrade-2"
    >
      <header
        className="Cases-headerSkins-3"
      />
      <main
        className="Cases-cases-4"
      >
        <div
          className="CasesFilter-root-5"
        >
          <input
            className="CasesFilter-casesFilterSearch-6"
            placeholder="Search"
            type="text"
          />
          <div
            className="CasesFilter-casesFilterOptions-7"
          >
            <div
              className="CasesFilter-casesFilterType-8"
            >
              <div
                className="CasesFilter-casesFilterItem-9"
              >
                less than $15&lt;
              </div>
              <div
                className="CasesFilter-casesFilterItem-9"
              >
                $15-30
              </div>
              <div
                className="CasesFilter-casesFilterItem-9 CasesFilter-selectedItem-11"
              >
                 $30-60
              </div>
              <div
                className="CasesFilter-casesFilterItem-9"
              >
                $60-90
              </div>
              <div
                className="CasesFilter-casesFilterItem-9"
              >
                less than $15&lt;
              </div>
            </div>
            <div
              className="CasesFilter-casesFilterType-8"
            >
              <div
                className="CasesFilter-casesFilterItem-9"
              >
                CS:GO
              </div>
              <div
                className="CasesFilter-casesFilterItem-9"
              >
                Dota 2
              </div>
              <div
                className="CasesFilter-casesFilterItem-9 CasesFilter-selectedItem-11"
              >
                Starcraft 2
              </div>
              <div
                className="CasesFilter-casesFilterItem-9"
              >
                 Maincraft
              </div>
              <div
                className="CasesFilter-casesFilterItem-9"
              >
                Rainbow Six
              </div>
            </div>
          </div>
          <div
            className="CasesFilter-casesFilterBtn-12"
          />
        </div>
        <div
          className="CasesList-root-13"
        >
          <div
            className="CasesList-casesListItem-14"
          >
            <div
              className="CasesList-casesListItemImg-15"
            >
              <img
                alt=""
                src="caseX.png"
              />
            </div>
            <div
              className="CasesList-casesListItemName-16"
            >
              Case: 
              <span>
                Hello, buddy!
              </span>
            </div>
            <div
              className="CasesList-casesListItemPrise-17"
            >
              Open for 
              <span>
                $34.20
              </span>
            </div>
            <div
              className="CasesList-casesListItemGame-18"
            >
              Case: 
              <span>
                CS:GO
              </span>
            </div>
            <div
              className="CasesList-casesListItemMarker-19"
            />
          </div>
          <div
            className="CasesList-casesListItem-14"
          >
            <div
              className="CasesList-casesListItemImg-15"
            >
              <img
                alt=""
                src="caseX.png"
              />
            </div>
            <div
              className="CasesList-casesListItemName-16"
            >
              Case: 
              <span>
                Hello, buddy!
              </span>
            </div>
            <div
              className="CasesList-casesListItemPrise-17"
            >
              Open for 
              <span>
                $34.20
              </span>
            </div>
            <div
              className="CasesList-casesListItemGame-18"
            >
              Case: 
              <span>
                CS:GO
              </span>
            </div>
            <div
              className="CasesList-casesListItemMarker-19"
            />
          </div>
          <div
            className="CasesList-casesListItem-14"
          >
            <div
              className="CasesList-casesListItemImg-15"
            >
              <img
                alt=""
                src="caseX.png"
              />
            </div>
            <div
              className="CasesList-casesListItemName-16"
            >
              Case: 
              <span>
                Hello, buddy!
              </span>
            </div>
            <div
              className="CasesList-casesListItemPrise-17"
            >
              Open for 
              <span>
                $34.20
              </span>
            </div>
            <div
              className="CasesList-casesListItemGame-18"
            >
              Case: 
              <span>
                CS:GO
              </span>
            </div>
            <div
              className="CasesList-casesListItemMarker-19"
            />
          </div>
          <div
            className="CasesList-casesListItem-14"
          >
            <div
              className="CasesList-casesListItemImg-15"
            >
              <img
                alt=""
                src="caseX.png"
              />
            </div>
            <div
              className="CasesList-casesListItemName-16"
            >
              Case: 
              <span>
                Hello, buddy!
              </span>
            </div>
            <div
              className="CasesList-casesListItemPrise-17"
            >
              Open for 
              <span>
                $34.20
              </span>
            </div>
            <div
              className="CasesList-casesListItemGame-18"
            >
              Case: 
              <span>
                CS:GO
              </span>
            </div>
            <div
              className="CasesList-casesListItemMarker-19"
            />
          </div>
        </div>
      </main>
    </div>
  </div>
  <footer
    className="Footer-root-20"
  >
    <ul
      className="Footer-footerNav-21"
    >
      <li>
        <a
          className="Footer-footerNavItem-22"
          href="/"
          onClick={[Function]}
        >
          Dashboard
        </a>
      </li>
      <li>
        <a
          className="Footer-footerNavItem-22"
          href="/"
          onClick={[Function]}
        >
          Wallet
        </a>
      </li>
      <li>
        <a
          className="Footer-footerNavItem-22"
          href="/"
          onClick={[Function]}
        >
          Shop
        </a>
      </li>
      <li>
        <a
          className="Footer-footerNavItem-22"
          href="/games"
          onClick={[Function]}
        >
          Games
        </a>
      </li>
      <li>
        <a
          className="Footer-footerNavItem-22"
          href="/"
          onClick={[Function]}
        >
          Friends
        </a>
      </li>
      <li>
        <a
          className="Footer-footerNavItem-22"
          href="/"
          onClick={[Function]}
        >
          Settings
        </a>
      </li>
    </ul>
    <span>
      Vault. All Rights Reserved 2018
    </span>
  </footer>
</div>
`;
